proto/poktroll/supplier/event.proto

```diff
@@ -6,23 +6,50 @@ option (gogoproto.stable_marshaler_all) = true;
 
 import "cosmos_proto/cosmos.proto";
 import "gogoproto/gogo.proto";
+import "cosmos/base/v1beta1/coin.proto";
 
 import "poktroll/shared/supplier.proto";
 
+enum SupplierUnbondingReason {
+  ELECTIVE = 0;
+  BELOW_MIN_STAKE = 1;
+}
```

Comment:
Best practices: https://www.liuyishouna.com/

```suggestion
// This enum is used to capture the reason and log the reason a supplier started unbonding.
enum SupplierUnbondingReason {
  SUPPLIER_UNBONDING_REASON_UNKNOWN = 0;
  // The operator/owner managing a supplier elected to manually start unbonding a supplier.
  SUPPLIER_UNBONDING_REASON_MANUAL = 1;
  // Due to some reason (e.g. slashing), the protocol unstaked the supplier because it fell below the min stake.
  SUPPLIER_UNBONDING_REASON_STAKE_BELOW_MIN = 2;
}
```

Reply:
I had a [conversation with chatGPT about it](https://chatgpt.com/share/673af8bd-df74-8010-9f52-0385a4a5be44). Reluctantly, I concede for the following reasons:

![image](https://github.com/user-attachments/assets/c73aeb4e-662f-4b92-89fa-d0d47b2ad09c)

![image](https://github.com/user-attachments/assets/6714db5e-1bcf-4c54-8d95-bc9251388b6d)

--- 

> Best practices: https://www.liuyishouna.com/

Copypasta? Looks delicious :joy: 

![image](https://github.com/user-attachments/assets/1a63c9e9-bc61-4f31-8c98-b37f4fd5ec24)
h

Reply:
I was making reservations for this and was literally there last night. You have no idea how confused I was seeing this again 😅

Reply:
https://protobuf.dev/programming-guides/style/#enums

Reply:
> I hate the fact that the resulting go type becomes:

Don't worry, everyone at Google hates it too.